tests/FunctionalConnectorTest.php

```diff
@@ -29,4 +34,174 @@ public function connectionToTcpServerShouldSucceedWithLocalhost()
         $connection->close();
         $server->close();
     }
+
+    /**
+     * @test
+     * @group internet
+     */
+    public function connectionToRemoteTCP4n6ServerShouldResultInOurIP()
+    {
+        $loop = Factory::create();
+
+        $connector = new Connector($loop, array('happy_eyeballs' => true));
+
+        $ip = Block\await($this->request('dual.tlund.se', $connector), $loop, self::TIMEOUT);
+
+        $this->assertSame($ip, filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_IPV6), $ip);
+    }
+
+    /**
+     * @test
+     * @group internet
+     */
+    public function connectionToRemoteTCP4ServerShouldResultInOurIP()
+    {
+        if ($this->ipv4() === false) {
+            $this->markTestSkipped('IPv4 not supported on this system');
+        }
+
+        $loop = Factory::create();
+
+        $connector = new Connector($loop, array('happy_eyeballs' => true));
+
+        $ip = Block\await($this->request('ipv4.tlund.se', $connector), $loop, self::TIMEOUT);
+
+        $this->assertSame($ip, filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4), $ip);
+        $this->assertFalse(filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6), $ip);
+    }
+
+    /**
+     * @test
+     * @group internet
+     */
+    public function connectionToRemoteTCP6ServerShouldResultInOurIP()
+    {
+        if ($this->ipv6() === false) {
+            $this->markTestSkipped('IPv6 not supported on this system');
+        }
+
+        $loop = Factory::create();
+
+        $connector = new Connector($loop, array('happy_eyeballs' => true));
+
+        $ip = Block\await($this->request('ipv6.tlund.se', $connector), $loop, self::TIMEOUT);
+
+        $this->assertFalse(filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4), $ip);
+        $this->assertSame($ip, filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6), $ip);
+    }
+
+    /**
+     * @test
+     * @group internet
+     *
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessageRegExp /Connection to ipv6.tlund.se:80 failed/
```

Comment:
This test expects an exception and has assertions, can you check which is correct?

Reply:
Whoops, resolved that :+1: